docs/dev.rst

```diff
@@ -1,20 +1,540 @@
+
+
+
+Skip ``unpackaged/*`` in Reference Repositories
+***********************************************
+
+If the referenced repository has dependency metadata under ``unpackaged/pre`` or ``unpackaged/post``, use the ``skip`` option to skip deploying that metadata with the dependency.
+
+    Example: Salesforce EDA
+
+.. code-block:: yaml
+
+    project:
+        dependencies:
+            - github: https://github.com/SalesforceFoundation/EDA
+              skip: unpackaged/post/course_connection_record_types
+
+
+Automatic Cleaning of ``meta.xml`` Files on Deploy
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To let CumulusCI fully manage the project's dependencies, the ``deploy`` task (and other tasks based on ``cumulusci.tasks.salesforce.Deploy``, or subclasses of it) automatically removes the ``<packageVersion>`` element and its children from all ``meta.xml`` files in the deployed metadata. Removing these elements does not affect the files on the filesystem.
+
+This feature supports CumulusCI's automatic dependency resolution by avoiding a need for projects to manually update XML files to reflect current dependency package versions.
+
+.. note:: If the metadata being deployed references namespaced metadata that does not exist in the currently installed package, the deployment throws an error as expected.
+
+The automatic cleaning of ``meta.xml`` files can be disabled by setting the ``clean_meta_xml`` option to ``False``. Prior to the addition of this functionality, there were unnecessary delays in the CumulusCI release cycle due to the need to create a new commit on ``main`` (and thus a feature branch, PR, code review, and so on) just to update the ``meta.xml`` files. CumulusCI's GitHub dependency functionality already handles requiring a new production release, so the only reason to do this commit was for the ``meta.xml`` files. Automatically cleaning the meta.xml files on deploy eliminates the need for this commit.
+
+One drawback of this approach is that developers need to handle the diffs in the ``meta.xml`` files by either ignoring them, or committing them as part of their work in a feature branch. 
+
+    Example: The diffs come from a scenario of Package B, which extends Package A. When a new production release of Package A is published, the ``update_dependencies`` task for Package B installs the new version. When metadata is then retrieved from the org, the ``meta.xml`` files reference the new version while the repository's ``meta.xml`` files reference an older version.
+
+    The main difference between this situation and one where the ``meta.xml`` file is automatically cleaned is that avoiding the diffs in ``meta.xml`` files is a convenience for developers rather than a requirement for builds and releases. 
```

Comment:
Upon second/third/fourth read, this sentence seems to get more and more confusing. What exactly is a diff?

Reply:
I'm adding this on something to revisit after the beta release (or potentially next week if I have time).

Reply:
But yes, this is very dense and not easy to comprehend.